src/main/scala/com/gu/ZuoraClient.scala

```diff
@@ -0,0 +1,289 @@
+  Host: rest.apisandbox.zuora.com
+  apiAccessKeyId: mario.galic_SB@guardian.co.uk
+  apiSecretAccessKey: ************
+  Accept: application/json
+  Content-Type: application/json
+  cache-control: no-cache
+  {
+      "remove": [
+          {
+              "ratePlanId": "2c92c0fb6736875d0167552b1a3a5058",
+              "contractEffectiveDate": "2019-03-07"
+          }
+      ],
+      "add": [
+          {
+              "productRatePlanId": "2c92c0f965dc30640165f150c0956859",
+              "contractEffectiveDate": "2019-03-07",
+              "chargeOverrides": [
+                  {
+                      "productRatePlanChargeId": "2c92c0f865d273010165f16ada0a4346",
+                      "price": 99
+                  }
+              ]
+          }
+      ]
+  }
+    */
+  def removeAndAddAProductRatePlan(
+      subscriptionName: String,
+      body: PriceRiseRequest): PriceRiseResponse = {
+    val response = Http(s"$host/v1/subscriptions/$subscriptionName")
+      .header("Authorization", s"Bearer $accessToken")
+      .header("content-type", "application/json")
+      .postData(write(body))
+      .method("PUT")
+      .asString
+
+    response.code match {
+      case 200 => parse(response.body).extract[PriceRiseResponse]
+      case _ => throw new RuntimeException(s"$subscriptionName failed to raise price due to Zuora networking issue: $response")
+    }
+  }
+
+  case class ExtendTerm(
+    currentTerm: String,
+    currentTermPeriodType: String
+  )
+
+  def extendTerm(
+    subscriptionName: String,
+    body: ExtendTerm): PriceRiseResponse = {
+    val response = Http(s"$host/v1/subscriptions/$subscriptionName")
+      .header("Authorization", s"Bearer $accessToken")
+      .header("content-type", "application/json")
+      .postData(write(body))
+      .method("PUT")
+      .asString
+
+    response.code match {
+      case 200 => parse(response.body).extract[PriceRiseResponse]
+      case _ => throw new RuntimeException(s"$subscriptionName failed to raise price due to Zuora networking issue: $response")
+    }
+  }
+}
+
+case class Token(
+  access_token: String,
+  token_type: String,
+  expires_in: String,
+  scope: String,
+  jti: String
+)
+
+// https://www.zuora.com/developer/api-reference/#operation/createToken
+object ZuoraOauth extends ZuoraJsonFormats {
+  import java.util.{Timer, TimerTask}
+  import ZuoraHostSelector._
+
+  var accessToken: String = null
+
+  private def getAccessToken(): String = {
+    val response = Http(s"$host/oauth/token")
+      .postForm(Seq(
+        "client_id" -> Config.Zuora.client_id,
+        "client_secret" -> Config.Zuora.client_secret,
+        "grant_type" -> "client_credentials"
+      ))
+      .asString
+      .body
+
+    parse(response).extract[Token].access_token
+  }
+
+  private val timer = new Timer()
+
+  timer.schedule(
+    new TimerTask { def run(): Unit = accessToken = getAccessToken() },
+    0, 55 * 60 * 1000 // refresh token every 55 min
```

Comment:
Do these expire hourly?

Reply:
Yes, we can see expiry in response of `POST /oauth/token`:

```
{
    "access_token": "*************",
    "token_type": "bearer",
    "expires_in": 3599,
    ...
}
```